configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -683,7 +683,20 @@ private static Object coerce(TypeToken<?> type, String value)
         if (valueOf != null) {
             if (type.isSubtypeOf(valueOf.getGenericReturnType())) {
                 try {
-                    return valueOf.invoke(null, value);
+                    return invokeFactory(valueOf, value);
+                }
+                catch (ReflectiveOperationException e) {
+                    return null;
+                }
+            }
+        }
+
+        // Look for a static of(String) method
+        Method of = stringAcceptingMethods.get("of");
+        if (of != null) {
```

Comment:
I think that for readability we can replace these with instanceof check like that:

```
       if (stringAcceptingMethods.get("of") instanceof Method of) {
            if (type.isSubtypeOf(of.getGenericReturnType())) {
                try {
                    return invokeFactory(of, value);
                }
                catch (ReflectiveOperationException e) {
                    return null;
                }
            }
        }
```

Reply:
This won't catch a null value. If you can apply this technique to other similar places in that class, that would be great.

Reply:
That's an interesting idea...

Reply:
(We don't have Error Prone here, though)